@xen-orchestra/fs/src/azure.js

```diff
@@ -114,10 +124,7 @@ export default class AzureHandler extends RemoteHandlerAbstract {
     for (let i = 0; i < blockCount; i++) {
       const blockId = Buffer.from(i.toString().padStart(6, '0')).toString('base64')
       blockIds.push(blockId)
-
-      const start = i * MAX_BLOCK_SIZE
-      const end = Math.min(start + MAX_BLOCK_SIZE, data.length)
-      const chunk = data.slice(start, end)
+      const chunk = await this.#getChunk(input, i, streamLength)
```

Comment:
please try to compute the minimal chunk size with : 
* must be bigger than the minimal chunk size accepted by azure
* must be smaller than the biggest chunk size accepted by azure
* must be big enough to send the full files when the file is split in the max number of chunk per file